Control/lib/middleware/detectorOwnership.middleware.js

```diff
@@ -0,0 +1,33 @@
+const {User} = require('../dtos/User');
+
+const {UnauthorizedAccessError} = require('./../errors/UnauthorizedAccessError.js');
+const {updateExpressResponseFromNativeError} = require('./../errors/updateExpressResponseFromNativeError.js');
+/**
+   * Middleware function to check detector ownership.
+   * Based on the session object, it checks if the user has ownership of the detector lock.
+   * @param {Request} req - Express Request object.
+   * @param {Response} res - Express Response object.
+   * @param {Function} next - Next middleware to call.
+   */
+const detectorOwnershipMiddleware = (req, res, next) => {
+  const { detectorId } = req.params;
+  const { name, username, personid, access } = req.session || {};
+
+  if (!detectorId || !access) {
+    updateExpressResponseFromNativeError(res, new UnauthorizedAccessError('Invalid request: missing information')); 
+  }
+  
+  try {
+    const user = new User(username, name, personid, access);
+    if (!user.belongsToDetector(detectorId)) {
```

Comment:
Hi @pepijndik . Very good first step for the PR!
There is one thing that we did not discuss and I did not explain in the ticket. We should take into consideration the scenario in which:
- a user with a role higher than `Detector` is requesting the lock. In this scenario, they will not have the `det-<name>` permission but instead they will have a higher role. In this case, we should ignore the check of this middleware function

I am thinking before this `if` condition, you have another one in which you check that the role of the user is sufficient to go `next()`. Have a look at the function `isRoleSufficient`. You should check that the current role of user is at least `Role.GLOBAL`. If so, then you do not need to check if the user belongs to the detector